assets/src/data/model/entity-factory/base-entity.js

```diff
@@ -0,0 +1,203 @@
+	 */
+	get saveState() {
+		return this[ PRIVATE_PROPERTIES.SAVE_STATE ];
+	}
+
+	/**
+	 * Whether the current save state is SAVE_STATE.NEW
+	 * @return {boolean}  True means SAVE_STATE.NEW is the save state.
+	 */
+	get isNew() {
+		return this.saveState === SAVE_STATE.NEW;
+	}
+
+	/**
+	 * Whether the current save state is SAVE_STATE.DIRTY
+	 * @return {boolean}  True means SAVE_STATE.DIRTY is the save state.
+	 */
+	get isDirty() {
+		return this.saveState === SAVE_STATE.DIRTY;
+	}
+
+	/**
+	 * Whether the current save state is SAVE_STATE.CLEAN
+	 * @return {boolean}  True means SAVE_STATE.CLEAN is the save state.
+	 */
+	get isClean() {
+		return this.saveState === SAVE_STATE.CLEAN;
+	}
+}
+
+/**
+ * A function that gives a class the provided name
+ * (and optionally extends the provided object).
+ * @param {string} name
+ * @param {Object} extendedClass
+ * @return {Function} A function
+ */
+const nameClass = ( name, extendedClass ) => {
+	return class extends extendedClass {
+		static get name() {
+			return name;
+		}
+	};
+};
+
+/**
+ * A factory for entity factories.
+ *
+ * Calling this returns an object of factory functions that instantiate an
+ * instance of a named Entity.  The modelName is used as the name for the new
+ * entity.
+ *
+ * Two methods are available on the object returned: `createNew` and
+ * `fromExisting`.
+ *
+ * @param {string} modelName  The model for the entity
+ * @param {Object} schema     The schema for the model. This is the schema
+ * provided by the OPTIONS endpoint for the model.
+ * @param {Array} fieldPrefixes An array of field prefixes for base fields on
+ * on the model (eg. Event model has `[ EVT ]` prefixes on fields, Datetime model
+ * has [ `DTT`, `DTT_EVT` ]
+ * @return {function(*=)} A factory for instantiating an entity instance.
+ */
+const createEntityFactory = ( modelName, schema, fieldPrefixes = [] ) => {
+	const Entity = nameClass(
+		upperFirst( camelCase( modelName ) ),
+		BaseEntity
+	);
+	return {
+		/**
+		 * Exposes modelName so client code can derive what model this factory
+		 * is for from any given factory.
+		 * @type string
+		 */
+		modelName: modelName,
+		/**
+		 * This is the class definition for the Entity.  Typically this is
+		 * retrieved for the ability to do instanceof checks.
+		 */
+		classDef: Entity,
+		/**
+		 * This returns an instance of Entity for the given arguments with the
+		 * indication this is a new non-persisted entity.  This means:
+		 *
+		 * - All field values are populated and any not provided will be
+		 *   populated with default values defined by the schema.
+		 * - Generates temporary unique ids for the primary key fields on the
+		 *   entity (using cuid).
+		 * - Sets the `isNew` flag to true for the entity so client code is able
+		 *   to discover which entities have never been persisted.
+		 * - Sets the `dirty` flag to true because the entity has never been
+		 *   persisted.
```

Comment:
ya the comment docs need fixed here a bit.  I just need to remove the line regarding dirty flag.  In earlier iterations there was both `isNew` and `isDirty` flags but thanks  to early feedback from you I realized it was unnecessary.   Note however that there is still a `isNew` argument flag in constructing `BaseEntity` because there needs to be a way to indicate this is a new entity (so construction follows a slightly different path).